src/main/java/org/jenkinsci/backend/confluence/pageremover/webapp/PageRemoverResource.java

```diff
@@ -0,0 +1,180 @@
+ * limitations under the License.
+ */
+package org.jenkinsci.backend.confluence.pageremover.webapp;
+
+import com.cybozu.labs.langdetect.LangDetectException;
+import com.cybozu.labs.langdetect.Language;
+import com.google.common.base.Optional;
+import hudson.plugins.jira.soap.RemotePage;
+import org.apache.commons.mail.Email;
+import org.apache.commons.mail.EmailException;
+import org.apache.commons.mail.SimpleEmail;
+import org.jenkinsci.backend.confluence.pageremover.Connection;
+import org.jenkinsci.backend.confluence.pageremover.LanguageDetection;
+import org.jenkinsci.backend.confluence.pageremover.PageNotification;
+import org.jenkinsci.backend.confluence.pageremover.Space;
+import org.jenkinsci.backend.ldap.AccountServer;
+import org.jenkinsci.backend.ldap.Config;
+import org.kohsuke.stapler.config.ConfigurationLoader;
+
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+import java.io.File;
+
+import static org.apache.commons.lang.exception.ExceptionUtils.getFullStackTrace;
+import static org.jenkinsci.backend.confluence.pageremover.Spambot.BLACKLIST;
+
+@Path("/page-remover")
+@Produces(MediaType.TEXT_PLAIN)
+public class PageRemoverResource {
+    private final String space;
+    private final String mailRecipient;
+    private final String smtpServer;
+
+    public PageRemoverResource(String space, String mailRecipient, String smtpServer) {
+        this.space = space;
+        this.mailRecipient = mailRecipient;
+        this.smtpServer = smtpServer;
+    }
+
+    @POST
+    public Response submit(@QueryParam("subject") Optional<String> subject,
+                           @QueryParam("body-plain") Optional<String> body) throws Exception {
+
+        PageNotification pageNotification = PageNotification.parse(subject.or(""), body.or(""));
+
+        if (subject.or("").startsWith("Re: ") && body.isPresent()) {
+            removePage(subject.get(), body.get());
+            return Response.ok().build();
+        }
+
+        if (pageNotification != null) {
+            processNotification(pageNotification);
+            return Response.ok().build();
+        }
+
+        System.err.println("Not a confluence notification email: " + subject.or("[empty subject]"));
+        return Response.ok().build();
+
+    }
+
+    /**
+     * Processes notification email from Confluence, and take appropriate actions.
+     */
+    private void processNotification(PageNotification n) throws Exception {
+        System.err.println("Parsed "+n);
+
+        if (n.action.equals("added")) {
+            Connection con = new Connection();
+            RemotePage p = con.getPage(n.spaceID, n.pageTitle);
+            Language lang = null;
+            try {
+                lang = new LanguageDetection().detect(p.getContent());
+                if (lang.lang.equals("id") && lang.prob>0.99) {
```

Comment:
I wonder (aloud) if making the spam probability configurable might be good to do. That way I can tweak that on the fly once its running if we find out the 0.99 probability is too high